### PR TITLE
Python. celery

### PR DESCRIPTION
**Pull request by AlanMamphs**

### CODE REVIEW

File: python/apps/utils/rabbitmq.py [hunk 1]
```diff
@@ -84,3 +84,73 @@ def call(self, queryset=None, serializer=None, exchange_name='', exchange_type='
...
+        self.q_receiving = q_receiving
+
+        self.channel.queue_declare(queue=self.q_receiving, durable=True)
+        self.channel.exchange_declare(exchange=exchange_name, exchange_type=exchange_type, durable=True)
+        if queryset and serializer:
+            message = self.message_to_server(queryset, serializer)
+            cor_id = json.loads(message)
+            uuid = cor_id['uuid']
+        else:
+            message = message
```
Comment (eamanbekov) [inline python/apps/utils/rabbitmq.py]: Did not get the meaning of this statement

File: python/apps/utils/rabbitmq.py [hunk 2]
```diff
@@ -84,3 +84,73 @@ def call(self, queryset=None, serializer=None, exchange_name='', exchange_type='
...
 
         return self.response
+
+    def on_response(self, ch, method, properties, body):
+        print('I have got the messsage from server')
+        self.response = body
+
+
+
+    def call_java(self, queryset=None, serializer=None, exchange_name='', exchange_type='topic', q_receiving='',
```
Comment (eamanbekov) [inline python/apps/utils/rabbitmq.py]: Cannot see any from `Rabbitmq` class, cause every time you call the method, you are re-declaring attributes. May be you should initialize them once and reuse them :+1:

File: python/apps/utils/rabbitmq.py [hunk 3]
```diff
@@ -84,3 +84,73 @@ def call(self, queryset=None, serializer=None, exchange_name='', exchange_type='
...
+             q_sending='', message=''):
+
+        self.response = None
+        self.q_sending = q_sending
+        self.q_receiving = q_receiving
```
Comment (eamanbekov) [inline python/apps/utils/rabbitmq.py]: You are assigning to objects variables, but I did not see any usage of these objects variables. E.g. in other class methods.

File: python/apps/vacancies/tasks.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+
+from djcelery import celery
+from apps.utils.rabbitmq import RabbitMQ
+from apps.departments.models import Department
+
+@celery.task
+def send_message_to_java(queryset, serializer):
+	rabbit = RabbitMQ(host="192.168.89.80", user="guest", password="guest")
```
[the remark this answers is not in the document]
Comment (AlanMamphs) [reply python/apps/vacancies/tasks.py]: It is according to the celery tutorial from multiple sources.

File: python/config/base.py [hunk 5]
```diff
@@ -39,8 +39,14 @@ class Base(Configuration):
...
         'apps.vacancies',
-        'apps.templates'
+        'apps.templates',
+
+        'djcelery',
     ]
+    import djcelery
+    djcelery.setup_loader()
+
+    BROKER_URL = "amqp://guest:guest@localhost:5672//"
```
[the remark this answers is not in the document]
Comment (AlanMamphs) [reply python/config/base.py]: It is according to the celery tutorial from multiple sources. Broker url is used by celery in the background.

File: python/apps/utils/rabbitmq.py (hunk 1 above)
Comment (AlanMamphs) [reply python/apps/utils/rabbitmq.py]: message variable is assigned to function input.

File: python/apps/utils/rabbitmq.py (hunk 2 above)
Comment (AlanMamphs) [reply python/apps/utils/rabbitmq.py]: I thought about this. Initializing them once is a good option, however there would be too many input variables while instantiating class. I decided to separate them using different methods.

File: python/apps/utils/rabbitmq.py (hunk 3 above)
Comment (AlanMamphs) [reply python/apps/utils/rabbitmq.py]: it is used in queue binding statement.

**Pull request merged**